result-async/src/main/scala/uscala/result/async/AsyncResult.scala

```diff
@@ -0,0 +1,89 @@
+package uscala.result.async
+
+import uscala.result.Result
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.Try
+
+final case class AsyncResult[+A, +B](underlying: Future[Result[A, B]]) {
```

Comment:
Not sure about it being a case class, given the implementation I would have probably chosen a "regular" class.